xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/macro/MacroId.java

```diff
@@ -106,15 +106,12 @@ public int hashCode()
     @Override
     public boolean equals(Object object)
     {
-        if (object == null) {
+        if (object == null || object.getClass() != getClass()) {
```

Comment:
Actually since MacroId is not really supposed to be extended you could replace this with a `if (!(object instanceof MacroId))`. Can't be null if it's an instance of MacroId.

Also I would put it after the `if (object == this)` test instead of before.

Reply:
Done